lib/jna/clink_jna.cc

```diff
@@ -108,15 +106,19 @@ SparseVectorJNA *OneHotEncoderModel_transform(clink::OneHotEncoderModel *model,
                                               const int value,
                                               const int columnIndex) {
   auto sparse_vector = model->transform(value, columnIndex);
-  CLINK_JNA_HANDLE_ERROR(sparse_vector.takeError())
+  CLINK_JNA_HANDLE_ERROR(sparse_vector)
+  if (sparse_vector.IsError()) {
+    return NULL;
+  }
```

Comment:
Could we change the return value type to `AsyncValueRef<SparseVectorJNA>` or simply `uniqe_ptr<SparseVectorJNA>`? In general we prefer to use reference-counted pointer to reduce the chance of avoid memory leak.

Same for every invocation of `new xxx`.

Reply:
As discussed offline, we plan to add functions to the Java part that tracks usages of such C++ objects, as the life span of these objects extends beyond the scope of C++ code into Java code. I'll first add a TODO here and provide further implementations afterwards.